skylighting-core/src/Skylighting/Tokenizer.hs

```diff
@@ -552,18 +550,11 @@ regExpr dynamic re inp = do
   -- return $! traceShowId $! (reStr, inp)
   let reStr = reString re
   when (BS.take 2 reStr == "\\b") $ wordBoundary inp
-  compiledREs <- gets compiledRegexes
-  regex <- case Map.lookup re compiledREs of
-              Nothing -> do
-                cre <- case compileRegex (reCaseSensitive re) reStr of
-                         Right r  -> return r
-                         Left e   -> throwError $
-                           "Error compiling regex " ++
-                            UTF8.toString reStr ++ ": " ++ e
-                modify $ \st -> st{ compiledRegexes =
-                      Map.insert re cre (compiledRegexes st) }
-                return cre
-              Just cre -> return cre
+  regex <- case compileRE re of
+            Right r  -> return r
+            Left e   -> throwError $
+              "Error compiling regex " ++
+              UTF8.toString reStr ++ ": " ++ e
   regex' <- if dynamic
```

Comment:
I see here that I was attempting to prevent repeated compilations by storing compiled REs in state.  Do you have a diagnosis for why that didn't work?

Reply:
So this does work, but unfortunately it's not enough. The tokeniser state is per-call to `tokenize`, which means the compiled regex is reused inside one code block, but is thrown away and not reused for the next code block we come to highlight.

Reply:
Ah right, thanks.